evap/staff/templates/staff_course_comments_quick.html

```diff
@@ -0,0 +1,436 @@
+
+                    this.slide("right", true);
+                    this.focusedButton = this.slider.find("[data-action=" + action + "]");
+                    this.focusedButton.focus();
+                    clearTimeout(this.buttonTimeout);
+                    this.buttonTimeout = setTimeout($.proxy(this.finishSlide, this), 200);
+                },
+
+                finishSlide: function() {
+                    this.updateState();
+                    this.focusedButton.blur();
+                },
+
+                repeat: function(action) {
+                    let reviewed = [];
+                    let unreviewed = [];
+
+                    this.answers.items.each(function() {
+                        let item = $(this);
+                        if(item.is("[data-review]")) {
+                            reviewed.push(item);
+                        } else {
+                            unreviewed.push(item);
+                        }
+                    });
+
+                    if(action !== "all" && unreviewed.length) {
+                        this.index = reviewed.length;
+                    } else {
+                        this.index = 0;
+                    }
+
+                    this.items = reviewed.concat(unreviewed);
+
+                    let direction = action === "start"? "right" : "left";
+                    this.answers.slide(this.active, direction);
+                    this.updateState();
+                },
+
+                updateState: function(noFooter) {
+                    this.slider.find(".slider-inner > :first-child").toggleClass("disabled", this.index === 0);
+                    this.slider.find(".slider-inner > :last-child").toggleClass("disabled", !this.isReviewing);
+
+                    this.slider.find("[data-counter=reviewed-left]").text(this.numReviewed);
+                    this.slider.find("[data-counter=unreviewed-left]").text(this.index - this.numReviewed);
+                    this.slider.find("[data-counter=reviewed-right]").text(this.numAlreadyReviewed);
+                    let unreviewed = this.items.length - this.index - this.numAlreadyReviewed - 1;
+                    this.slider.find("[data-counter=unreviewed-right]").text(Math.max(unreviewed, 0));
+
+                    if(!this.isReviewing) {
+                        let reviewed = this.numReviewed === this.index;
+                        this.alert.toggleClass("alert-warning", !reviewed);
+                        this.alert.toggleClass("alert-success", reviewed);
+                        this.alert.find("[data-content=unreviewed]").toggleClass("d-none", reviewed);
+                        this.alert.find("[data-content=reviewed]").toggleClass("d-none", !reviewed);
+                        this.slider.find("[data-action=repeat-unreviewed]").toggleClass("d-none", reviewed);
+                    }
+
+                    if(noFooter) {
+                        return;
+                    }
```

Comment:
why is this early return (and the noFooter parameter) necessary?

Reply:
okay, that was not obvious to me. then why is `updateState` called without `noFooter=True` in `press`? 

also, i assume this is the purpose of the button timeout. would it be more appropriate to start that timeout inside updateState if noFooter was set?

Reply:
...i just noticed that `press` calls `updateState` and then `slide`, which also calls `updateState`.

Reply:
Yes, that's correct, as the button needs to be coloured before sliding to the next answer. I'll just collect all this to find a semantically better version